grid_tools/man/man3/pietc.3
.TH "pietc" 3 "Tue Jan 21 2025" "Version 1.13.0" "grid_tools" \" -*- nroff -*-
.ad l
.nh
.SH NAME
pietc \- Some of the commonly used constants (pi etc) mainly for double-precision subroutines\&.  

.SH SYNOPSIS
.br
.PP
.SS "Variables"

.in +1c
.ti -1c
.RI "complex(dpc), parameter \fBc0\fP =(\fBu0\fP, \fBu0\fP)"
.br
.RI "complex zero "
.ti -1c
.RI "complex(dpc), parameter \fBc1\fP =(\fBu1\fP, \fBu0\fP)"
.br
.RI "complex one "
.ti -1c
.RI "complex(dpc), parameter \fBci\fP =(\fBu0\fP, \fBu1\fP)"
.br
.RI "complex i=sqrt(-1) "
.ti -1c
.RI "complex(dpc), parameter \fBcipi\fP =\fBci\fP*\fBpi\fP"
.br
.RI "complex i*pi "
.ti -1c
.RI "real(dp), parameter \fBdtor\fP =\fBpi\fP/180"
.br
.RI "Degrees to radians\&. "
.ti -1c
.RI "real(dp), parameter \fBeuler\fP =0\&.57721566490153286060651209008240243104215933593992359880e0_dp"
.br
.RI "Euler-Mascheroni constant\&. "
.ti -1c
.RI "logical, parameter \fBf\fP =\&.false\&."
.br
.RI "for pain-relief in logical ops "
.ti -1c
.RI "complex(dpc), parameter \fBmc1\fP =\-\fBc1\fP"
.br
.RI "complex minus-one "
.ti -1c
.RI "complex(dpc), parameter \fBmci\fP =\-\fBci\fP"
.br
.RI "complex minus-i "
.ti -1c
.RI "real(dp), parameter \fBmo2\fP =\-\fBo2\fP"
.br
.RI "minus-half "
.ti -1c
.RI "real(dp), parameter \fBmo3\fP =\-\fBo3\fP"
.br
.RI "minus-third "
.ti -1c
.RI "real(dp), parameter \fBmo4\fP =\-\fBo4\fP"
.br
.RI "minus-fourth "
.ti -1c
.RI "real(dp), parameter \fBmo5\fP =\-\fBo5\fP"
.br
.RI "minus-fifth "
.ti -1c
.RI "real(dp), parameter \fBmo6\fP =\-\fBo6\fP"
.br
.RI "minus-sixth "
.ti -1c
.RI "real(dp), parameter \fBms10\fP =\-\fBs10\fP"
.br
.RI "minus-sine(10 deg) "
.ti -1c
.RI "real(dp), parameter \fBms11\fP =\-\fBs11\fP"
.br
.RI "minus-sine(90/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBms13\fP =\-\fBs13\fP"
.br
.RI "minus-sine(90/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms15\fP =\-\fBs15\fP"
.br
.RI "minus-sine(15 deg) "
.ti -1c
.RI "real(dp), parameter \fBms18\fP =\-\fBs18\fP"
.br
.RI "minus-sine(18 deg) "
.ti -1c
.RI "real(dp), parameter \fBms20\fP =\-\fBs20\fP"
.br
.RI "minus-sine(20 deg) "
.ti -1c
.RI "real(dp), parameter \fBms22\fP =\-\fBs22\fP"
.br
.RI "minus-sine(22\&.5 deg) "
.ti -1c
.RI "real(dp), parameter \fBms26\fP =\-\fBs26\fP"
.br
.RI "minus-sine(90*2/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms30\fP =\-\fBs30\fP"
.br
.RI "minus-sine(30 deg) "
.ti -1c
.RI "real(dp), parameter \fBms34\fP =\-\fBs34\fP"
.br
.RI "minus-sine(90*3/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBms36\fP =\-\fBs36\fP"
.br
.RI "minus-sine(36 deg) "
.ti -1c
.RI "real(dp), parameter \fBms39\fP =\-\fBs39\fP"
.br
.RI "minus-sine(90*3/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms40\fP =\-\fBs40\fP"
.br
.RI "minus-sine(40 deg) "
.ti -1c
.RI "real(dp), parameter \fBms45\fP =\-\fBs45\fP"
.br
.RI "minus-sine(45 deg) "
.ti -1c
.RI "real(dp), parameter \fBms50\fP =\-\fBs50\fP"
.br
.RI "minus-sine(50 deg) "
.ti -1c
.RI "real(dp), parameter \fBms51\fP =\-\fBs51\fP"
.br
.RI "minus-sine(90*4/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms54\fP =\-\fBs54\fP"
.br
.RI "minus-sine(54 deg) "
.ti -1c
.RI "real(dp), parameter \fBms56\fP =\-\fBs56\fP"
.br
.RI "minus-sine(90*5/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBms60\fP =\-\fBs60\fP"
.br
.RI "minus-sine(60 deg) "
.ti -1c
.RI "real(dp), parameter \fBms64\fP =\-\fBs64\fP"
.br
.RI "minus-sine(90*5/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms68\fP =\-\fBs68\fP"
.br
.RI "minus-sine(67\&.5 deg) "
.ti -1c
.RI "real(dp), parameter \fBms70\fP =\-\fBs70\fP"
.br
.RI "minus-sine(70 deg) "
.ti -1c
.RI "real(dp), parameter \fBms72\fP =\-\fBs72\fP"
.br
.RI "minus-sine(72 deg) "
.ti -1c
.RI "real(dp), parameter \fBms75\fP =\-\fBs75\fP"
.br
.RI "minus-sine(75 deg) "
.ti -1c
.RI "real(dp), parameter \fBms77\fP =\-\fBs77\fP"
.br
.RI "minus-sine(90*6/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBms79\fP =\-\fBs79\fP"
.br
.RI "minus-sine(90*7/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBms80\fP =\-\fBs80\fP"
.br
.RI "minus-sine(80 deg) "
.ti -1c
.RI "real(dp), parameter \fBmu1\fP =\-\fBu1\fP"
.br
.RI "minus-one "
.ti -1c
.RI "real(dp), parameter \fBmu2\fP =\-\fBu2\fP"
.br
.RI "minus-two "
.ti -1c
.RI "real(dp), parameter \fBmu3\fP =\-\fBu3\fP"
.br
.RI "minus-three "
.ti -1c
.RI "real(dp), parameter \fBmu4\fP =\-\fBu4\fP"
.br
.RI "minus-four "
.ti -1c
.RI "real(dp), parameter \fBmu5\fP =\-\fBu5\fP"
.br
.RI "minus-five "
.ti -1c
.RI "real(dp), parameter \fBmu6\fP =\-\fBu6\fP"
.br
.RI "minus-six "
.ti -1c
.RI "real(dp), parameter \fBo2\fP =\fBu1\fP/\fBu2\fP"
.br
.RI "half "
.ti -1c
.RI "real(dp), parameter \fBo3\fP =\fBu1\fP/\fBu3\fP"
.br
.RI "third "
.ti -1c
.RI "real(dp), parameter \fBo4\fP =\fBu1\fP/\fBu4\fP"
.br
.RI "fourth "
.ti -1c
.RI "real(dp), parameter \fBo5\fP =\fBu1\fP/\fBu5\fP"
.br
.RI "fifth "
.ti -1c
.RI "real(dp), parameter \fBo6\fP =\fBu1\fP/\fBu6\fP"
.br
.RI "sixth "
.ti -1c
.RI "real(dp), parameter \fBor2\fP =\fBu1\fP/\fBr2\fP"
.br
.RI "1\&. "
.ti -1c
.RI "real(dp), parameter \fBor3\fP =\fBu1\fP/\fBr3\fP"
.br
.RI "1\&. "
.ti -1c
.RI "real(dp), parameter \fBor5\fP =\fBu1\fP/\fBr5\fP"
.br
.RI "1\&. "
.ti -1c
.RI "real(dp), parameter \fBphi\fP =1\&.6180339887498948482045868343656381177203091798057628621354e0_dp"
.br
.RI "Golden number\&. "
.ti -1c
.RI "real(dp), parameter \fBpi\fP =3\&.1415926535897932384626433832795028841971693993751058209749e0_dp"
.br
.RI "Pi\&. "
.ti -1c
.RI "real(dp), parameter \fBpi2\fP =6\&.2831853071795864769252867665590057683943387987502116419498e0_dp"
.br
.RI "Pi*2\&. "
.ti -1c
.RI "real(dp), parameter \fBpih\fP =1\&.5707963267948966192313216916397514420985846996875529104874e0_dp"
.br
.RI "pi*half "
.ti -1c
.RI "real(dp), parameter \fBr2\fP =1\&.4142135623730950488016887242096980785696718753769480731766e0_dp"
.br
.RI "Square root of 2\&. "
.ti -1c
.RI "real(dp), parameter \fBr3\fP =1\&.7320508075688772935274463415058723669428052538103806280558e0_dp"
.br
.RI "Square root of 3\&. "
.ti -1c
.RI "real(dp), parameter \fBr5\fP =2\&.2360679774997896964091736687312762354406183596115257242708e0_dp"
.br
.RI "Square root of 5\&. "
.ti -1c
.RI "real(dp), parameter \fBrpi\fP =1\&.7724538509055160272981674833411451827975494561223871282138e0_dp"
.br
.RI "square root of pi "
.ti -1c
.RI "real(dp), parameter \fBrtod\fP =180/\fBpi\fP"
.br
.RI "radians to degrees "
.ti -1c
.RI "real(dp), parameter \fBs10\fP =\&.173648177666930348851716626769314796000375677184069387236241e0_dp"
.br
.RI "sine(10 deg) "
.ti -1c
.RI "real(dp), parameter \fBs11\fP =\&.195090322016128267848284868477022240927691617751954807754502e0_dp"
.br
.RI "sine(90/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBs13\fP =\&.222520933956314404288902564496794759466355568764544955311987e0_dp"
.br
.RI "sine(90/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBs15\fP =\&.258819045102520762348898837624048328349068901319930513814003e0_dp"
.br
.RI "sine(15 deg) "
.ti -1c
.RI "real(dp), parameter \fBs18\fP =\&.309016994374947424102293417182819058860154589902881431067724e0_dp"
.br
.RI "sine(18 deg) "
.ti -1c
.RI "real(dp), parameter \fBs20\fP =\&.342020143325668733044099614682259580763083367514160628465048e0_dp"
.br
.RI "sine(20 deg) "
.ti -1c
.RI "real(dp), parameter \fBs22\fP =\&.382683432365089771728459984030398866761344562485627041433800e0_dp"
.br
.RI "sine(22\&.5 deg) "
.ti -1c
.RI "real(dp), parameter \fBs26\fP =\&.433883739117558120475768332848358754609990727787459876444547e0_dp"
.br
.RI "sine(90*2/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBs30\fP =\fBo2\fP"
.br
.RI "sine(30 deg) "
.ti -1c
.RI "real(dp), parameter \fBs34\fP =\&.555570233019602224742830813948532874374937190754804045924153e0_dp"
.br
.RI "sine(90*3/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBs36\fP =\&.587785252292473129168705954639072768597652437643145991072272e0_dp"
.br
.RI "sine(36 deg) "
.ti -1c
.RI "real(dp), parameter \fBs39\fP =\&.623489801858733530525004884004239810632274730896402105365549e0_dp"
.br
.RI "sine(90*3/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBs40\fP =\&.642787609686539326322643409907263432907559884205681790324977e0_dp"
.br
.RI "sine(40 deg) "
.ti -1c
.RI "real(dp), parameter \fBs45\fP =\fBor2\fP"
.br
.RI "sine(45 deg) "
.ti -1c
.RI "real(dp), parameter \fBs50\fP =\&.766044443118978035202392650555416673935832457080395245854045e0_dp"
.br
.RI "sine(50 deg) "
.ti -1c
.RI "real(dp), parameter \fBs51\fP =\&.781831482468029808708444526674057750232334518708687528980634e0_dp"
.br
.RI "sine(90*4/7 deg) "
.ti -1c
.RI "real(dp), parameter \fBs54\fP =\&.809016994374947424102293417182819058860154589902881431067724e0_dp"
.br
.RI "sine(54 deg) "
.ti -1c
.RI "real(dp), parameter \fBs56\fP =\&.831469612302545237078788377617905756738560811987249963446124e0_dp"
.br
.RI "sine(90*5/8 deg) "
.ti -1c
.RI "real(dp), parameter \fBs60\fP =\fBr3\fP*\fBo2\fP"
.br
.RI "sine(60 deg) "
.ti -1c
.RI "real(dp), parameter \fBs64\fP =\&.900968867902419126236102319507445051165919162131857150053562e0_dp"
.br
.RI "sine(90*5/7 deg) "
.ti -1c

sorc/grid_tools.fd/regional_esg_grid.fd/pietc.f90
!> @file
!! @brief Double-precision constants.
!! @author R. J. Purser @date 2014

!> Some of the commonly used constants (pi etc) mainly for double-precision
!! subroutines.
!!
!! ms10 etc are needed to satisfy the some (eg., gnu fortran) compilers'
!! more rigorous standards regarding the way "data" statements are initialized.
!! Zero and the first few units are u0,u1,u2, etc., their reciprocals being,
!! o2,o3 etc and their square roots, r2,r3. Reciprocal roots are or2,or3 etc.
!!
!! @author R. J. Purser @date 2014
module pietc
use pkind, only: dp,dpc
implicit none
logical ,parameter:: T=.true. !< for pain-relief in logical ops
logical ,parameter:: F=.false. !< for pain-relief in logical ops
real(dp),parameter:: u0=0_dp !< zero
real(dp),parameter:: u1=1_dp !< one
real(dp),parameter:: mu1=-u1 !< minus-one
real(dp),parameter:: u2=2_dp !< two
real(dp),parameter:: mu2=-u2 !< minus-two
real(dp),parameter:: u3=3_dp !< three
real(dp),parameter:: mu3=-u3 !< minus-three
real(dp),parameter:: u4=4_dp !< four
real(dp),parameter:: mu4=-u4 !< minus-four
real(dp),parameter:: u5=5_dp !< five
real(dp),parameter:: mu5=-u5 !< minus-five
real(dp),parameter:: u6=6_dp !< six
real(dp),parameter:: mu6=-u6 !< minus-six
real(dp),parameter:: o2=u1/u2 !< half
real(dp),parameter:: o3=u1/u3 !< third
real(dp),parameter:: o4=u1/u4 !< fourth
real(dp),parameter:: o5=u1/u5 !< fifth
real(dp),parameter:: o6=u1/u6 !< sixth
real(dp),parameter:: mo2=-o2 !< minus-half
real(dp),parameter:: mo3=-o3 !< minus-third
real(dp),parameter:: mo4=-o4 !< minus-fourth
real(dp),parameter:: mo5=-o5 !< minus-fifth
real(dp),parameter:: mo6=-o6 !< minus-sixth
real(dp),parameter:: pi =3.1415926535897932384626433832795028841971693993751058209749e0_dp !< Pi.
real(dp),parameter:: pi2=6.2831853071795864769252867665590057683943387987502116419498e0_dp !< Pi*2.
real(dp),parameter:: pih=1.5707963267948966192313216916397514420985846996875529104874e0_dp !< pi*half
real(dp),parameter:: rpi=1.7724538509055160272981674833411451827975494561223871282138e0_dp !< square root of pi
real(dp),parameter:: r2 =1.4142135623730950488016887242096980785696718753769480731766e0_dp !< Square root of 2.
real(dp),parameter:: r3 =1.7320508075688772935274463415058723669428052538103806280558e0_dp !< Square root of 3.
real(dp),parameter:: r5 =2.2360679774997896964091736687312762354406183596115257242708e0_dp !< Square root of 5.
real(dp),parameter:: or2=u1/r2 !< 1./root of two
real(dp),parameter:: or3=u1/r3 !< 1./root of three
real(dp),parameter:: or5=u1/r5 !< 1./root of five
real(dp),parameter:: phi=1.6180339887498948482045868343656381177203091798057628621354e0_dp !< Golden number.
real(dp),parameter:: euler=0.57721566490153286060651209008240243104215933593992359880e0_dp !< Euler-Mascheroni constant.
real(dp),parameter:: dtor=pi/180 !< Degrees to radians
real(dp),parameter:: rtod=180/pi !< radians to degrees
! Sines of all main fractions of 90 degrees (down to ninths): 
real(dp),parameter:: s10=.173648177666930348851716626769314796000375677184069387236241e0_dp !< sine(10 deg)
real(dp),parameter:: s11=.195090322016128267848284868477022240927691617751954807754502e0_dp !< sine(90/8 deg)
real(dp),parameter:: s13=.222520933956314404288902564496794759466355568764544955311987e0_dp !< sine(90/7 deg)
real(dp),parameter:: s15=.258819045102520762348898837624048328349068901319930513814003e0_dp !< sine(15 deg)
real(dp),parameter:: s18=.309016994374947424102293417182819058860154589902881431067724e0_dp !< sine(18 deg)
real(dp),parameter:: s20=.342020143325668733044099614682259580763083367514160628465048e0_dp !< sine(20 deg)
real(dp),parameter:: s22=.382683432365089771728459984030398866761344562485627041433800e0_dp !< sine(22.5 deg)
real(dp),parameter:: s26=.433883739117558120475768332848358754609990727787459876444547e0_dp !< sine(90*2/7 deg)
real(dp),parameter:: s30=o2 !< sine(30 deg)
real(dp),parameter:: s34=.555570233019602224742830813948532874374937190754804045924153e0_dp !< sine(90*3/8 deg)
real(dp),parameter:: s36=.587785252292473129168705954639072768597652437643145991072272e0_dp !< sine(36 deg)
real(dp),parameter:: s39=.623489801858733530525004884004239810632274730896402105365549e0_dp !< sine(90*3/7 deg)
real(dp),parameter:: s40=.642787609686539326322643409907263432907559884205681790324977e0_dp !< sine(40 deg)
real(dp),parameter:: s45=or2 !< sine(45 deg)
real(dp),parameter:: s50=.766044443118978035202392650555416673935832457080395245854045e0_dp !< sine(50 deg)
real(dp),parameter:: s51=.781831482468029808708444526674057750232334518708687528980634e0_dp !< sine(90*4/7 deg)
real(dp),parameter:: s54=.809016994374947424102293417182819058860154589902881431067724e0_dp !< sine(54 deg)
real(dp),parameter:: s56=.831469612302545237078788377617905756738560811987249963446124e0_dp !< sine(90*5/8 deg)
real(dp),parameter:: s60=r3*o2 !< sine(60 deg)
real(dp),parameter:: s64=.900968867902419126236102319507445051165919162131857150053562e0_dp !< sine(90*5/7 deg)
real(dp),parameter:: s68=.923879532511286756128183189396788286822416625863642486115097e0_dp !< sine(67.5 deg)
real(dp),parameter:: s70=.939692620785908384054109277324731469936208134264464633090286e0_dp !< sine(70 deg)
real(dp),parameter:: s72=.951056516295153572116439333379382143405698634125750222447305e0_dp !< sine(72 deg)
real(dp),parameter:: s75=.965925826289068286749743199728897367633904839008404550402343e0_dp !< sine(75 deg)
real(dp),parameter:: s77=.974927912181823607018131682993931217232785800619997437648079e0_dp !< sine(90*6/7 deg)
real(dp),parameter:: s79=.980785280403230449126182236134239036973933730893336095002916e0_dp !< sine(90*7/8 deg)
real(dp),parameter:: s80=.984807753012208059366743024589523013670643251719842418790025e0_dp !< sine(80 deg)
! ... and their minuses:
real(dp),parameter:: ms10=-s10 !< minus-sine(10 deg)
real(dp),parameter:: ms11=-s11 !< minus-sine(90/8 deg)
real(dp),parameter:: ms13=-s13 !< minus-sine(90/7 deg)
real(dp),parameter:: ms15=-s15 !< minus-sine(15 deg)
real(dp),parameter:: ms18=-s18 !< minus-sine(18 deg)
real(dp),parameter:: ms20=-s20 !< minus-sine(20 deg)
real(dp),parameter:: ms22=-s22 !< minus-sine(22.5 deg)
real(dp),parameter:: ms26=-s26 !< minus-sine(90*2/7 deg)
real(dp),parameter:: ms30=-s30 !< minus-sine(30 deg)
real(dp),parameter:: ms34=-s34 !< minus-sine(90*3/8 deg)
real(dp),parameter:: ms36=-s36 !< minus-sine(36 deg)
real(dp),parameter:: ms39=-s39 !< minus-sine(90*3/7 deg)
real(dp),parameter:: ms40=-s40 !< minus-sine(40 deg)
real(dp),parameter:: ms45=-s45 !< minus-sine(45 deg)
real(dp),parameter:: ms50=-s50 !< minus-sine(50 deg)
real(dp),parameter:: ms51=-s51 !< minus-sine(90*4/7 deg)
real(dp),parameter:: ms54=-s54 !< minus-sine(54 deg)
real(dp),parameter:: ms56=-s56 !< minus-sine(90*5/8 deg)
real(dp),parameter:: ms60=-s60 !< minus-sine(60 deg)
real(dp),parameter:: ms64=-s64 !< minus-sine(90*5/7 deg)
real(dp),parameter:: ms68=-s68 !< minus-sine(67.5 deg)
real(dp),parameter:: ms70=-s70 !< minus-sine(70 deg)
real(dp),parameter:: ms72=-s72 !< minus-sine(72 deg)
real(dp),parameter:: ms75=-s75 !< minus-sine(75 deg)
real(dp),parameter:: ms77=-s77 !< minus-sine(90*6/7 deg)
real(dp),parameter:: ms79=-s79 !< minus-sine(90*7/8 deg)
real(dp),parameter:: ms80=-s80 !< minus-sine(80 deg)
complex(dpc),parameter:: c0=(u0,u0) !< complex zero
complex(dpc),parameter:: c1=(u1,u0) !< complex one
complex(dpc),parameter:: mc1=-c1 !< complex minus-one
complex(dpc),parameter:: ci=(u0,u1) !< complex i=sqrt(-1)
complex(dpc),parameter:: mci=-ci !< complex minus-i
complex(dpc),parameter:: cipi=ci*pi !< complex i*pi
! Main fractional rotations, as unimodular complex numbers: 
complex(dpc),parameter:: z000=c1 !< exp(2*pi*i*0)
complex(dpc),parameter:: z010=( s80,s10) !< exp(2*pi*i/36)
complex(dpc),parameter:: z011=( s79,s11) !< exp(2*pi*i/32)
complex(dpc),parameter:: z013=( s77,s13) !< exp(2*pi*i/28)
complex(dpc),parameter:: z015=( s75,s15) !< exp(2*pi*i/24)
complex(dpc),parameter:: z018=( s72,s18) !< exp(2*pi*i/20)
complex(dpc),parameter:: z020=( s70,s20) !< exp(2*pi*i/18)
complex(dpc),parameter:: z022=( s68,s22) !< exp(2*pi*i/16)
complex(dpc),parameter:: z026=( s64,s26) !< exp(2*pi*i/14)
complex(dpc),parameter:: z030=( s60,s30) !< exp(2*pi*i/12)
complex(dpc),parameter:: z034=( s56,s34) !< exp(2*pi*i*3/32)
complex(dpc),parameter:: z036=( s54,s36) !< exp(2*pi*i/10)
complex(dpc),parameter:: z039=( s51,s39) !< exp(2*pi*i*3/28)
complex(dpc),parameter:: z040=( s50,s40) !< exp(2*pi*i/9)
complex(dpc),parameter:: z045=( s45,s45) !< exp(2*pi*i/8)
complex(dpc),parameter:: z050=( s40,s50) !< exp(2*pi*i*5/36)
complex(dpc),parameter:: z051=( s39,s51) !< exp(2*pi*i/7)
complex(dpc),parameter:: z054=( s36,s54) !< exp(2*pi*i*3/20)
complex(dpc),parameter:: z056=( s34,s56) !< exp(2*pi*i*5/32)
complex(dpc),parameter:: z060=( s30,s60) !< exp(2*pi*i/6)
complex(dpc),parameter:: z064=( s26,s64) !< exp(2*pi*i*5/28)
complex(dpc),parameter:: z068=( s22,s68) !< exp(2*pi*i*3/16)
complex(dpc),parameter:: z070=( s20,s70) !< exp(2*pi*i*7/36)
complex(dpc),parameter:: z072=( s18,s72) !< exp(2*pi*i/5)
complex(dpc),parameter:: z075=( s15,s75) !< exp(2*pi*i*5/24)
complex(dpc),parameter:: z077=( s13,s77) !< exp(2*pi*i*3/14)
complex(dpc),parameter:: z079=( s11,s79) !< exp(2*pi*i*7/32)
complex(dpc),parameter:: z080=( s10,s80) !< exp(2*pi*i*2/9)
complex(dpc),parameter:: z090=ci !< exp(2*pi*i/4)
complex(dpc),parameter:: z100=(ms10,s80) !< exp(2*pi*i*5/18)
complex(dpc),parameter:: z101=(ms11,s79) !< exp(2*pi*i*9/32)
complex(dpc),parameter:: z103=(ms13,s77) !< exp(2*pi*i*2/7)
complex(dpc),parameter:: z105=(ms15,s75) !< exp(2*pi*i*7/24)
complex(dpc),parameter:: z108=(ms18,s72) !< exp(2*pi*i*3/10)
complex(dpc),parameter:: z110=(ms20,s70) !< exp(2*pi*i*11/36)
complex(dpc),parameter:: z112=(ms22,s68) !< exp(2*pi*i*5/16)
complex(dpc),parameter:: z116=(ms26,s64) !< exp(2*pi*i*9/28)
complex(dpc),parameter:: z120=(ms30,s60) !< exp(2*pi*i/3)
complex(dpc),parameter:: z124=(ms34,s56) !< exp(2*pi*i*11/32)
complex(dpc),parameter:: z126=(ms36,s54) !< exp(2*pi*i*7/20)
complex(dpc),parameter:: z129=(ms39,s51) !< exp(2*pi*i*5/14)
complex(dpc),parameter:: z130=(ms40,s50) !< exp(2*pi*i*13/36)
complex(dpc),parameter:: z135=(ms45,s45) !< exp(2*pi*i*3/8)
complex(dpc),parameter:: z140=(ms50,s40) !< exp(2*pi*i*7/18)
complex(dpc),parameter:: z141=(ms51,s39) !< exp(2*pi*i*11/28)
complex(dpc),parameter:: z144=(ms54,s36) !< exp(2*pi*i*2/5)
complex(dpc),parameter:: z146=(ms56,s34) !< exp(2*pi*i*13/32)
complex(dpc),parameter:: z150=(ms60,s30) !< exp(2*pi*i*5/12)
complex(dpc),parameter:: z154=(ms64,s26) !< exp(2*pi*i*3/7)
complex(dpc),parameter:: z158=(ms68,s22) !< exp(2*pi*i*7/16)
complex(dpc),parameter:: z160=(ms70,s20) !< exp(2*pi*i*4/9)
complex(dpc),parameter:: z162=(ms72,s18) !< exp(2*pi*i*9/20)
complex(dpc),parameter:: z165=(ms75,s15) !< exp(2*pi*i*11/24)
complex(dpc),parameter:: z167=(ms77,s13) !< exp(2*pi*i*13/28)
complex(dpc),parameter:: z169=(ms79,s11) !< exp(2*pi*i*15/32)
complex(dpc),parameter:: z170=(ms80,s10) !< exp(2*pi*i*17/36)
complex(dpc),parameter:: z180=-z000 !< exp(2*pi*i/2)
complex(dpc),parameter:: z190=-z010 !< exp(2*pi*i*19/36)
complex(dpc),parameter:: z191=-z011 !< exp(2*pi*i*17/32)
complex(dpc),parameter:: z193=-z013 !< exp(2*pi*i*15/28)
complex(dpc),parameter:: z195=-z015 !< exp(2*pi*i*13/24)
complex(dpc),parameter:: z198=-z018 !< exp(2*pi*i*11/20)
complex(dpc),parameter:: z200=-z020 !< exp(2*pi*i*5/9)
complex(dpc),parameter:: z202=-z022 !< exp(2*pi*i*9/16)
complex(dpc),parameter:: z206=-z026 !< exp(2*pi*i*4/7)
complex(dpc),parameter:: z210=-z030 !< exp(2*pi*i*7/12)
complex(dpc),parameter:: z214=-z034 !< exp(2*pi*i*19/32)
complex(dpc),parameter:: z216=-z036 !< exp(2*pi*i*3/5)
complex(dpc),parameter:: z219=-z039 !< exp(2*pi*i*17/28)
complex(dpc),parameter:: z220=-z040 !< exp(2*pi*i*11/18)
complex(dpc),parameter:: z225=-z045 !< exp(2*pi*i*5/8)
complex(dpc),parameter:: z230=-z050 !< exp(2*pi*i*23/36)
complex(dpc),parameter:: z231=-z051 !< exp(2*pi*i*9/14)
complex(dpc),parameter:: z234=-z054 !< exp(2*pi*i*13/20)
complex(dpc),parameter:: z236=-z056 !< exp(2*pi*i*21/32)
complex(dpc),parameter:: z240=-z060 !< exp(2*pi*i*2/3)
complex(dpc),parameter:: z244=-z064 !< exp(2*pi*i*19/28)
complex(dpc),parameter:: z248=-z068 !< exp(2*pi*i*11/16)
complex(dpc),parameter:: z250=-z070 !< exp(2*pi*i*25/36)
complex(dpc),parameter:: z252=-z072 !< exp(2*pi*i*7/10)
complex(dpc),parameter:: z255=-z075 !< exp(2*pi*i*17/24)
complex(dpc),parameter:: z257=-z077 !< exp(2*pi*i*5/7)
complex(dpc),parameter:: z259=-z079 !< exp(2*pi*i*23/32)
complex(dpc),parameter:: z260=-z080 !< exp(2*pi*i*13/18)
complex(dpc),parameter:: z270=-z090 !< exp(2*pi*i*3/4)
complex(dpc),parameter:: z280=-z100 !< exp(2*pi*i*7/9)
complex(dpc),parameter:: z281=-z101 !< exp(2*pi*i*25/32)
complex(dpc),parameter:: z283=-z103 !< exp(2*pi*i*11/14)
complex(dpc),parameter:: z285=-z105 !< exp(2*pi*i*19/24)
complex(dpc),parameter:: z288=-z108 !< exp(2*pi*i*4/5)
complex(dpc),parameter:: z290=-z110 !< exp(2*pi*i*29/36)
complex(dpc),parameter:: z292=-z112 !< exp(2*pi*i*13/16)
complex(dpc),parameter:: z296=-z116 !< exp(2*pi*i*23/28)
complex(dpc),parameter:: z300=-z120 !< exp(2*pi*i*5/6)
complex(dpc),parameter:: z304=-z124 !< exp(2*pi*i*27/32)
complex(dpc),parameter:: z306=-z126 !< exp(2*pi*i*17/20)
complex(dpc),parameter:: z309=-z129 !< exp(2*pi*i*6/7)
complex(dpc),parameter:: z310=-z130 !< exp(2*pi*i*31/36)
complex(dpc),parameter:: z315=-z135 !< exp(2*pi*i*7/8)
complex(dpc),parameter:: z320=-z140 !< exp(2*pi*i*8/9)
complex(dpc),parameter:: z321=-z141 !< exp(2*pi*i*25/28)
complex(dpc),parameter:: z324=-z144 !< exp(2*pi*i*9/10)
complex(dpc),parameter:: z326=-z146 !< exp(2*pi*i*29/32)
complex(dpc),parameter:: z330=-z150 !< exp(2*pi*i*11/12)
complex(dpc),parameter:: z334=-z154 !< exp(2*pi*i*13/14)
complex(dpc),parameter:: z338=-z158 !< exp(2*pi*i*15/16)
complex(dpc),parameter:: z340=-z160 !< exp(2*pi*i*17/18)
complex(dpc),parameter:: z342=-z162 !< exp(2*pi*i*19/20)
complex(dpc),parameter:: z345=-z165 !< exp(2*pi*i*23/24)
complex(dpc),parameter:: z347=-z167 !< exp(2*pi*i*27/28)
complex(dpc),parameter:: z349=-z169 !< exp(2*pi*i*31/32)
complex(dpc),parameter:: z350=-z170 !< exp(2*pi*i*35/36)
end module pietc
